src/Entity/EntityClassDiagram.puml

```diff
@@ -0,0 +1,105 @@
+@startuml
+class Player {
+// manual assigning all atr
+// also load sprites
+// and putting sfxs
+- constructor(TileMap)
+==
+// loop through enemies
+// check check attacks
+// check if intersects
++ checkAttack(ArrayList<Enemy> enemy): void
++ hit(int damage): void
++ getNextPosition(): void
++ update(): void
++ draw():void @override
++ getter()
+--ENUMS--
+int IDLE = 0
+int WALKING = 1
+int JUMPING = 2
+int FALLING = 3
+int GLIDING = 4
```

Comment:
I think you can just remove the gliding state, the player doesn't have any wings

Reply:
working on it!

Reply:
resolved